### DISCUSSION TITLE
Point me in a direction for desk accessories?

### DISCUSSION THREAD

**Discussion by tschak909** [autc04/Retro68#229] [Q&A]
Hey guys, can someone point me in a direction with the toolbox manuals etc to be able to do desk accessories?

-Thom

**Comment by autc04** [comment]
First, desk accessories are code resources (see the SystemExtension sample) but with a twist.
The 'DRVR' starts with a data structure that contains offsets to the actual code.
That means, you will probably have to fiddle with some direct GNU Assembler directives to get the layout right, and you will have more than one "main" program, but rather several entry points which may be called more than once.
You should probably also read https://github.com/autc04/Retro68/wiki/Global-Variables-and-Relocation-(68K).

The original docs on how to write Desk Accessories are in the "Desk Manager" chapter of Inside Macintosh (Volume One). By the time the "new" Inside Macintosh was written in the early nineties, desk accessories were officially obsolete, so I don't think there's a new chapter on them.

**Comment by tschak909** [comment]
As it happens, one of my long time friends is in the middle of doing a new ethernet card for PDS macs, and needed to write a new DRVR, and worked out most of the details in his repo, 

https://github.com/rhalkyard/SEthernet/tree/main/software/driver :)

maybe we can fold some things upstream to make lives easier for the next crazy person who wants to do something like this? :)

-Thom
